### PR TITLE
Add show method and test for GitTree

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/tree.jl [hunk 1]
```diff
@@ -70,3 +70,9 @@ function Base.show(io::IO, te::GitTreeEntry)
     println(io, "Entry type: ", entrytype(te))
     println(io, "Entry OID: ", entryid(te))
 end
+
+function Base.show(io::IO, tree::GitTree)
+    println(io, "GitTree:")
+    println(io, "Owner: ", repository(tree))
```
Comment (ararslan) [inline base/libgit2/tree.jl]: Is there a nice `show` method for `GitRepo` that we should take advantage of here?

**Comment by kshyatt** [comment]
What's the deal with the OSX failure? Not my fault, I think?

**Comment by ararslan** [comment]
I think the error is #17626.

**Comment by kshyatt** [comment]
So is this mergeable?

**Review by ararslan — APPROVED** [review]

**Pull request merged**